### PR TITLE
SEN-2612 Add recipes for Hamcrest's hasItem assertion.

### PR DESCRIPTION
**Pull request by bbogason**

### CODE REVIEW

**Review by wanderneto — APPROVED** [review]
Great work!
Everything works fine. 
Just one recipe is really complex but I'm sure that is the only way to create it.

Super minor: commit comment is saying that task is SEN-2612 when task is SEN-2614

**Comment by bbogason** [comment]
> Super minor: commit comment is saying that task is SEN-2612 when task is SEN-2614

Nice catch! (and I don't agree that it is super minor 😅). Fixed!

**Comment by seanf** [comment]
>> Super minor: commit comment is saying that task is SEN-2612 when task is SEN-2614

>Nice catch! (and I don't agree that it is super minor 😅). Fixed! 

@bbogason 
I guess we could rename the PR too, even if it has already been merged.

File: recipes/Java/AssertJ/From Hamcrest/Hamcrest_hasItem.yaml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                            reference:
+                              matches: org.hamcrest.Matcher<.*>
+                            checkInheritance: true
+                    argCount: 1
+          argCount: 2
+  actions:
+  - rewrite:
+      to: org.assertj.core.api.Assertions.assertThat({{{ arguments.0 }}}).contains({{{ arguments.1.arguments }}})
+- doStaticImports: true
+  name: '[direct] Change to assertThat(collection).as(reason).contains(expectedElement)'
```
Comment (camac) [inline recipes/Java/AssertJ/From Hamcrest/Hamcrest_hasItem.yaml:152]: Just checking are the **[direct]** and **[level1]** prefixes in these fix names meant to be retained? I am unsure what they mean so I am just wondering if they were used during recipe creation to check which fixes were showing and maybe the intention was to remove these before publishing the recipes?
They only seem to be in this recipe

**Comment by bbogason** [comment]
> I guess we could rename the PR too, even if it has already been merged.

Thanks! I meant to do that, but forgot. Fixed now.

File: recipes/Java/AssertJ/From Hamcrest/Hamcrest_hasItem.yaml (hunk 1 above)
Comment (bbogason) [reply recipes/Java/AssertJ/From Hamcrest/Hamcrest_hasItem.yaml:152]: Great catch! They were indeed only there for my own clarity while coming up with this recipe, and was not meant to be in the final version. Will fix!

**Pull request merged**